canonicalize.go

```diff
@@ -47,60 +99,23 @@ func (a attrsByKey) Less(i, j int) bool {
 	return composeAttr(a[i].Space, a[i].Key) < composeAttr(a[j].Space, a[j].Key)
 }
 
-func _canonicalHack(el *etree.Element, seenSoFar map[string]struct{}) *etree.Element {
```

Comment:
This function got moved down below and renamed to `canonicalPrep`. Otherwise it remains identical.